<!-- -->
<launch>
    
    <node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode" output="screen">
	  <param name="serial_port"         type="string" value="/dev/ttyUSB1"/>
	  <param name="serial_baudrate"     type="int"    value="115200"/><!--A1/A2 -->
	  <!--param name="serial_baudrate"     type="int"    value="256000"--><!--A3 -->
	  <param name="frame_id"            type="string" value="laser"/>
	  <param name="inverted"            type="bool"   value="false"/>
	  <param name="angle_compensate"    type="bool"   value="true"/>
    </node>        
	
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find rplidar_ros)/rviz/rplidar.rviz" />	

    
    <node name="obstacle_avoiding_node" pkg="obstacle_avoiding" type="obstacle_avoiding_node" output="screen">
        <param name="avoiding_disThreshold" value="2"/>
		<param name="speed" value="1.0"/>
		<param name="steeringAngle" value="15"/>
    </node>


    <node type="obstacle_detect_node" pkg ="obstacle_detect" name="obstacle_detect_node" output="screen"/>

    <node type="decision_making_node" pkg ="decision_making" name="decision_making_node" output="screen"/>
    

     
    <node name="path_tracking_node" pkg="path_tracking" type="path_tracking_node" output="screen">
		<param name="path_points_file" value="$(find record_gps_data)/../gps_data/gps_data.txt"/>
        <param name="vehicle_axis_dis" value="0.65"/>
		<param name="disThreshold" value="3.0"/>
		<param name="speed" value="1.0"/>    </node>

	<node name="nuogeng_node" pkg="nuogeng" type ="nuogeng_node" output="screen" >
	
		<param name ="port_name"   value="/dev/ttyUSB0"/>
		<param name ="baudrate"    value="115200"/>
	</node>
	



    <node type="base_control_node" pkg ="base_control" name="base_control_node" >
    	<param name ="stm32_port_name"   value="/dev/ttyTHS2"/>
    </node>

      
</launch>



